src/tokenizer/CMakeLists.txt

```diff
@@ -0,0 +1,38 @@
+set(TOKENIZER_ROOT ${CMAKE_CURRENT_SOURCE_DIR} PARENT_SCOPE)
+file(GLOB tokenizer_srcs CONFIGURE_DEPENDS
+        "${CMAKE_CURRENT_SOURCE_DIR}/*.cc"
+        "${CMAKE_CURRENT_SOURCE_DIR}/utils/*.cc"
+        "${CMAKE_CURRENT_SOURCE_DIR}/c_api/*.cc"
+)
+FetchContent_Declare(spm URL https://github.com/google/sentencepiece/archive/refs/tags/v0.1.99.zip)
+set(SPM_USE_EXTERNAL_ABSL OFF CACHE BOOL "Use external absl")
```

Comment:
Please note that this line of code means spm will bring in its own absl and protofbuf, and in this program no other part will need to use the two dependencies, otherwise there will be a conflict.